hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManager.java

```diff
@@ -169,7 +171,16 @@ public void addConnectionListener(ConnectionListener listener) {
 
     @Override
     public synchronized void accept(Packet packet) {
-        serverControl.process(packet);
+        // the packet was sent from 3.12 if the 4_0 flag is missing
+        // 4.0.1 and 4.2 members do not set this flag
```

Comment:
Do we need #17756 here as well? ([ref](https://github.com/hazelcast/hazelcast/pull/17702/files#r502042078))

Reply:
It's already part of this PR. Basically any instance starting with 4.0.2 (https://github.com/hazelcast/hazelcast/pull/16993) and continuing with 4.1 (https://github.com/hazelcast/hazelcast/pull/17756) and onwards will set the flag, so that this "migration" member will correctly interpret the message. Or am I missing something?

Reply:
To be more precise, it's already part of the base 4.1.1-migration branch, so this PR does not need to introduce it.

Reply:
Just wanted to double-check we don't have a regression on an earlier finding. Seen that it's on the `4.1.z` branch, so that's just fine.